### PR TITLE
#964 Right / left pages borders too thin and shadow not visible

### PR DESCRIPTION
**Pull request by amynbe**
Fix by forcing right/left page borders width to 2mm

### CODE REVIEW

**Comment by amynbe** [comment]
Does this look reasonable @ahmedre ?

**Comment by ahmedre** [comment]
jazakAllah khairan - the code looks fine, but want to run it and look at it first, will do so tomorrow in sha' Allah.

in the meanwhile, was curious as to why 2mm instead of a fixed number of dp instead?

**Comment by ahmedre** [comment]
it looks good in light mode - maybe in dark mode we can improve it some - @ozbek can you try this out and tell me what you think also?

File: app/src/main/java/com/quran/labs/androidquran/widgets/QuranPageLayout.java [hunk 1]
```diff
@@ -141,16 +143,26 @@ protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
     }
   }
 
+  private int getBorderWidth(Drawable drawable) {
+    if (drawable == lineDrawable)
+      return lineDrawable.getIntrinsicWidth();
+
+    // convert 2mm to pixels
+    return (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_MM,
+        2f,
```
Comment (ozbek) [inline app/src/main/java/com/quran/labs/androidquran/widgets/QuranPageLayout.java:152]: What's the rationale for using exactly 2mm? Why not 1mm for example?

Comment (amynbe) [reply app/src/main/java/com/quran/labs/androidquran/widgets/QuranPageLayout.java:152]: It seems to match the official screenshots https://android.quran.com/images/scr_2.png .
Smaller wouldn't achieve the desired effect (i.e let the brain know which side you're on).

To answer @ahmedre 's question I wasn't aware that dp was a resolution independent size so can replace mm with the equivalent dp if that's preferred/more efficient.

**Comment by ozbek** [comment]
Right border does not look very good in translation mode (text sticks very much to the edge). Can you take a look?

Before | After
------------ | -------------
before | after

**Comment by amynbe** [comment]
This fixes it @ozbek 

| Left  | Right |
| ------------- | ------------- |
| Screenshot_1590955063  | Screenshot_1590953739  |

**Comment by amynbe** [comment]
Can you give this a hopefully final review please when you get a chance @ozbek .

**Comment by ozbek** [comment]
LGTM, but brother @ahmedre should approve.

**Comment by ahmedre** [comment]
jazakumAllah khairan @amynbe and sorry it took so long to get this merged

**Comment by ahmedre** [comment]
and jazakumAllah khairan @ozbek for reviewing and trying it - i have gotten behind on PRs in the past few months but will try to be a bit faster in sha' Allah.

**Comment by amynbe** [comment]
Thank you Ahmed and Shuhrat, may Allah reward your efforts.

**Pull request merged**